src/odemis/acq/align/autofocus.py

```diff
@@ -834,6 +843,11 @@ def updateProgress(subf, start, end):
         ret = {}
         logging.debug("Running AutoFocusSpectrometer on %s, using %s, for the detectors %s, and using selector %s",
                       spgr, focuser, dets, selector)
+        # make sure the slit-in position is at the minimum when using an external spectrograph
+        if spgr.role == 'spectrograph-dedicated':
+            if spgr.position.value["slit-in"] != spgr.axes["slit-in"].range[0]:
+                spgr.moveAbsSync({"slit-in": spgr.axes["slit-in"].range[0]})
+
```

Comment:
The idea of the `spec-focus` alignment mode is that this sort of things is handled in the optical path manager. That's why there is a different mode from the "mirror-align". Similarly, here, instead of "tunnel-lens-align", I think the proper way is to have a "spec-ext-focus" mode in the path.py, which sets the hardware to the right position. I think it's mostly a copy of "tunnel-lens-align", but the slit-in is at 10e-6 m (like for "spec-focus").

Note this means you have to update the functions in the sparc2_align_tab, onAutoFocus() & onManualFocus().

Reply:
I think I misunderstood 'focus_mode' and 'alignment_mode' which are essentially the same but named differently.  Added spec-focus-ext to path.py.